src/QsCompiler/Transformations/ContentLifting.cs

```diff
@@ -64,17 +63,22 @@ public CallableDetails(QsCallable callable)
 
         public class TransformationState
         {
-            public bool IsValidScope = true;
-            public List<QsCallable> GeneratedOperations = null;
-            public ImmutableArray<LocalVariableDeclaration<NonNullable<string>>> GeneratedOpParams =
+            // ToDo: It should be possible to make these three properties private, 
+            // if we absorb the corresponding logic into LiftBody. 
+            internal bool IsValidScope = true;
+            internal bool ContainsParamRef = false;
+            internal ImmutableArray<LocalVariableDeclaration<NonNullable<string>>> GeneratedOpParams =
                 ImmutableArray<LocalVariableDeclaration<NonNullable<string>>>.Empty;
-            public bool ContainsParamRef = false;
 
-            public CallableDetails CurrentCallable = null;
-            public bool InBody = false;
-            public bool InAdjoint = false;
-            public bool InControlled = false;
-            public bool InWithinBlock = false;
+            internal CallableDetails CurrentCallable = null;
+
+            protected internal bool InBody = false;
+            protected internal bool InAdjoint = false;
+            protected internal bool InControlled = false;
+            protected internal bool InControlledAdjoint = false;
+            protected internal bool InWithinBlock = false;
```

Comment:
Since combining access modifiers in C# can result in some confusing behavior (confusing to me at least 😁), I just want to double check this does what you intend. My understanding is that this will make these fields accessible to anything in the assembly, _and_ _also_ accessible to any derived class outside of the assembly. Is that what you are going for here?

Reply:
It was indeed what I was going for. I did debate internally what they should be, and I could go either way, but then I ultimately didn't want to make them fully public (I can be convinced though).